### ISSUE TITLE
Categorize devices

### ISSUE THREAD

**Issue by sergio-domingues** [jas-/node-libnmap#58]
Is it possible to have a description of the device associated to each IP address? When one access the web interface of a local router it is available a list of devices with the device name and IP, it would be useful to access this information. Thanks in advance!

**Comment by jas-** [comment]
Like the OS and/or device type? Have you read the README regarding examples provided? It has already been provided.

**Comment by sergio-domingues** [comment]
Yes, I mean the OS and device type. I had only traversed the reports and didn't found any reference to those parameters. Now I've checked the flags.js example, which has the OS flag but did not find anything about the device type. I was trying to run this example locally but it fails, I'm guessing it's due to be running it on windows. Is it only supported on linux/osx ?

**Comment by jas-** [comment]
@sergio-domingues 
Telling me that the example fails doesn't give me much to go on. What failed? Did you receive and error message? Did you provide the necessary administrative credentials prior to running the scan?

Perhaps I should provide some resources that are readily available to you already that perhaps you are unaware of with regards to the user manual for the nmap tool. [nmap OS detection](https://nmap.org/book/man-os-detection.html), [nmap OS detection usage and examples](https://nmap.org/book/osdetect-usage.html) & [how to harden your OS against nmap OS detection](https://nmap.org/misc/defeat-nmap-osdetect.html)
